File: corpus/typescript/fixture_03.ts
type CartItem3 = { price: number; qty?: number };
export function calculateDiscount3(cartItems: CartItem3[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 30 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook3 {
  lookupPrice(sku: string): number { return sku.length + 3; }
}

File: corpus/go/fixture_03.go
package fixture3

type PriceBook3 struct {
    Multiplier float64
}

func CalculateDiscount3(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 30.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}
